Point_set_processing_3/include/CGAL/poisson_eliminate.h

```diff
@@ -0,0 +1,449 @@
+
+   \tparam OutputIterator Type of the output iterator. Must accept input of the same type as the iterator of `PointRange`.
+
+   \param points input point range
+   \param number_of_points target output size, must be smaller that the number of points in the input range
+   \param output where output points are put
+   \param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+
+   \cgalNamedParamsBegin
+       \cgalParamNBegin{point_map}
+         \cgalParamDescription{a property map associating points to the elements of the point set `points`}
+         \cgalParamType{a model of `ReadablePropertyMap` whose key type is the value type
+                        of the iterator of `PointRange` and whose value type is `geom_traits::Point_3`}
+         \cgalParamDefault{`CGAL::Identity_property_map<geom_traits::Point_3>`}
+       \cgalParamNEnd
+
+       \cgalParamNBegin{dimension}
+         \cgalParamDescription{The sampling domain of `points`, e.g., 2 if the points have been sampled from a 2d surface.}
+         \cgalParamType{unsigned integer}
+         \cgalParamDefault{2}
+       \cgalParamNEnd
+
+       \cgalParamNBegin{progressive}
+         \cgalParamDescription{reorders the points in `output` in a progressive way, i.e., the first n points in `output` with n < `number_of_points` have a poisson disk distribution with a larger radius. }
+         \cgalParamType{Boolean}
+         \cgalParamDefault{`false`}
+       \cgalParamNEnd
+
+       \cgalParamNBegin{maximum_radius}
+         \cgalParamDescription{radius of the poisson disk in which the neighboring points are taken into account for elimination.}
+         \cgalParamType{double}
+         \cgalParamDefault{the default is calculated from the `dimension`, the volume of the bounding box and the `number_of_points`. For more details, see parameter section in \ref Point_set_processing_3PoissonElimination.}
+       \cgalParamNEnd
+
+       \cgalParamNBegin{weight_function}
+         \cgalParamDescription{a weight function that calculates the weight of a neighbor point based on its squared distance and the `maximum_radius`.}
+         \cgalParamType{an instance of `std::function<double(const Point&, const Point&, double,double)>`.}
+         \cgalParamDefault{See parameter section in \ref Point_set_processing_3PoissonElimination.}
+       \cgalParamNEnd
+
+       \cgalParamNBegin{geom_traits}
+         \cgalParamDescription{an instance of a geometric traits class}
+         \cgalParamType{a model of `Kernel`}
+         \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+       \cgalParamNEnd
+     \cgalNamedParamsEnd
+*/
+template<class PointRange, class OutputIterator, class NamedParameters = parameters::Default_named_parameters>
+void poisson_eliminate(PointRange &points, std::size_t number_of_points, OutputIterator output, const NamedParameters& np = parameters::default_values()) {
```

Comment:
I wonder if eliminate is the right word since output points are copied. Also, why is `PointRange` not const? is it reordered?

Reply:
Changed to const